Distraction Shield/tooltipPage/tooltip.js

```diff
@@ -0,0 +1,85 @@
+
+var bg = chrome.extension.getBackgroundPage();
+
+var auth = bg.auth;
+
+var saveButton = $('#saveBtn');
+var sessionBtn = $('#sessionBtn');
+var optionsButton = $('#optionsBtn');
+var statisticsButton = $('#statisticsBtn');
+
+saveCurrentPageToBlacklist = function() {
+    chrome.tabs.query({active: true, currentWindow: true}, function (arrayOfTabs) {
+        var activeTab = arrayOfTabs[0];
+        bg.addUrlToBlockedSites(activeTab.url, setSaveButtonToSuccess);
+    });
+};
+
+setSaveButtonToSuccess = function () {
+    saveButton.attr('class', 'btn btn-success');
+    saveButton.html('Successfully added!');
+    setTimeout(function () {
+        saveButton.attr('class', 'btn btn-info');
+        saveButton.html(' Save current page ');
+    }, 4000);
+};
+
+redirectToStatistics = function() {
+    chrome.tabs.create({'url': chrome.runtime.getURL('statisticsPage/statistics.html')});
+};
+
+openOptionsPage = function() {
+    chrome.tabs.create({'url': chrome.runtime.getURL('optionsPage/options.html')});
+};
+
+redirectToLogin = function() {
+    chrome.tabs.create({'url': chrome.runtime.getURL('loginPage/login.html')});
+};
+
+logout = function () {
+    auth.logout().then(function () {
+        updateSessionbutton();
+    }, function () {
+        updateSessionbutton();
```

Comment:
I assume this calls updateSessionbutton either on success or failure of the logout and then this function in turn checks the state of the authentication - very convoluted application flow